rclcpp/include/rclcpp/publisher_options.hpp

```diff
@@ -64,8 +64,12 @@ struct PublisherOptionsBase
 template<typename Allocator>
 struct PublisherOptionsWithAllocator : public PublisherOptionsBase
 {
+  static_assert(
+    std::is_void<typename std::allocator_traits<Allocator>::value_type>::value,
+    "Publisher allocator value type must be void");
```

Comment:
why?
could't it be rebinded to `void`?

Reply:
It can be, but it's unnecessarily surprising. There's no setter API, `allocator` is a plain member-field. So a non-`void` allocator instance set will never be used, and we're forced to keep the allocator that is actually used separately. 

Also, what would be the point of allowing custom allocators of arbitrary type T? What's the use of accepting e.g. `std::allocator<std::vector<double>>`?

Reply:
Sounds good, I guess this doesn't break any existing code ...

Reply:
I'm personally fine with this check. If anyone was passing something different before, it wasn't doing what they expected anyways.